File: src/main/scala/com/bonc/Main/ModelsFactory.scala
package com.bonc.Main

import java.io.{File, FileInputStream}
import java.util.Properties

import com.bonc.interfaceRaw.IModel

//import com.bonc.Interface.IModel
import org.apache.spark.sql.Row

/**
  * Created by sid on 2018/7/23 0023.
  */

//见https://www.cnblogs.com/AaronCui/p/4915055.html
object ModelsFactory {
  def init(propsPath:String):Properties={
    var props = new Properties();
    props.load(new FileInputStream(new File(propsPath)))
    props
  }

  def produce(shortName:String,props:Properties):IModel[Row]={
//    var iModel: IModel[Row] = IModel[_]
    val fullName = props.getProperty(shortName.toLowerCase)
    val iModel:IModel[Row] = Class.forName(fullName).newInstance().asInstanceOf[IModel[Row]]
    iModel
  }

  def main(args: Array[String]): Unit = {
//    val prop = ModelsFactory.init("conf/default.properties")
//    val alias = prop.getProperty("TE")
//    println(alias)
//    val path = "conf/default.properties"
//    val dirs  = new File("")
//    println(dirs.getAbsolutePath+File.separator+path)



  }



}

File: src/main/java/com/bonc/JavaUtils/ModelsFactory.java
package com.bonc.JavaUtils;

import com.bonc.Interface.IModel;

import java.io.File;
import java.io.FileInputStream;
import java.io.IOException;
import java.util.Properties;

/**
 * Created by Administrator on 2018/7/23 0023.
 */
public class ModelsFactory {

    public static Properties init(String propsFilePath){
        Properties properties = new Properties();
        File file = new File(propsFilePath);
        try {
            properties.load(new FileInputStream(file));
            System.out.println("props:"+properties.getProperty("TE"));
        } catch (IOException e) {
            e.printStackTrace();
            System.err.println("Missing Property file!");
        }
        return properties;
    }

    public static IModel produce(String shortName,Properties props){
        IModel model = null;
        try {
            String modelName = props.getProperty(shortName);
            model = (IModel) Class.forName(modelName).newInstance();
        } catch (InstantiationException e) {
            e.printStackTrace();
        } catch (IllegalAccessException e) {
            e.printStackTrace();
        } catch (ClassNotFoundException e) {
            e.printStackTrace();
        }
        return model;
    }

    public static void main(String[] args) {
        String path = "conf/default.properties";
        Properties props = ModelsFactory.init(path);
        IModel te1 = ModelsFactory.produce("TE", props);
        System.out.println("jdj");
//        te1.setParam("s");


    }

}
